echopop/statistics.py

```diff
@@ -0,0 +1,268 @@
+        # ---- Sum of all weighted squared deviations
+        squared_deviation_wgt = (stratified_weights**2 * squared_deviation).sum(axis=1)
+        # ---- Compute the variance by incorporating the degrees of freedom
+        variance_arr[:, j - 1] = squared_deviation_wgt / sample_dof[j]
+
+    # Calculate the area-weight variance of the resulting transect-length-weighted biological
+    # variable
+    # ---- Variance
+    replicate_variance = (total_transect_area.to_numpy() ** 2 * variance_arr).sum(axis=1)
+    # ---- Convert to standard deviation
+    replicate_stdev = np.sqrt(replicate_variance)
+    # ---- Calculate the replicate mean
+    replicate_mean = (total_transect_area.to_numpy() * mean_arr).sum(axis=1)
+    # ---- Convert to CV
+    replicate_cv = replicate_stdev / replicate_mean
+
+    # Calculate the replicate grand total of the (weighted) biological variable
+    # ---- Overall mean of transect integrated sums
+    replicate_sum_mean = (total_transect_area.to_numpy() * total_arr).sum(
+        axis=1
+    ) / total_transect_area.sum()
+
+    # ---- Overall total using the overall mean
+    replicate_total = total_arr.sum(axis=1)
+
+    # Output the related summary statistics
+    # ---- Save the output resampled distributions
+    resampled_distributions = pd.DataFrame(
+        {
+            "realization": np.arange(1, transect_replicates + 1),
+            "mean_standardized": replicate_mean,
+            "standard_deviation": replicate_stdev,
+            "variance": replicate_variance,
+            "cv": replicate_cv,
+            "mean": replicate_sum_mean,
+            "total": replicate_total,
+        }
+    )
+    # ---- Save the stratified results
+    stratified_results = {
+        "variable": settings_dict["variable"],
+        "ci_percentile": 0.95,
+        "num_transects": strata_summary["transect_count"].sum(),
+        "total_area": strata_summary["transect_area_total"].sum(),
+        "mean": {
+            "weighted_estimate": np.mean(replicate_mean),
+            "unweighted_estimate": np.mean(replicate_sum_mean),
+            "weighted_confidence_interval": confidence_interval(replicate_mean),
+            "unweighted_confidence_interval": confidence_interval(replicate_sum_mean),
+        },
+        "variance": {
+            "estimate": np.mean(replicate_variance),
+            "confidence_interval": confidence_interval(replicate_variance),
+        },
+        "cv": {
+            "estimate": np.mean(replicate_cv),
+            "confidence_interval": confidence_interval(replicate_cv),
+        },
+        "total": {
+            "estimate": np.mean(replicate_total),
+            "confidence_interval": confidence_interval(replicate_total),
+        },
+    }
+    # ---- Return outputs
+    return resampled_distributions, stratified_results
+
+
+def array_math(left_array: np.ndarray, right_array: np.ndarray, operation: str) -> np.ndarray:
```

Comment:
I would suggest removing this function and just write out the transpose needed in the code. One-off utility functions like this makes code harder to read and are prone to error, because you are essentially trying to overload binary operators in a somewhat limited way. This is also only used twice and the underlying operations are very simple.